Perl cpi

notjeian opened the pull request:
Took long enough, but here it is. Works, as best I can tell.

notjeian committed:
working widechodes. finally.

fidsah commented on cpi/perl/Cocktography.pm:
```diff
@@ -0,0 +1,310 @@
+	my $dechoded;
+	my $prev_type = -1;
+	my $condom;
+
```
```suggestion
my $strokes;
```

fidsah commented on cpi/perl/Cocktography.pm:
```diff
@@ -0,0 +1,310 @@
+		
+		if ($cockblock->{'TYPE'} == $COCKBLOCK_TYPE{'SINGLETON'} && ($prev_type != $COCKBLOCK_TYPE{'INITIAL'} && $prev_type != $COCKBLOCK_TYPE{'INTERMEDIATE'})) {
+			my ($destroked, $strokecount) = $self->destroke($cyphallic);
+			$dechoded .= $destroked;
```
```suggestion
                        $strokes = $strokecount;
			$dechoded .= $destroked;
```

fidsah commented on cpi/perl/Cocktography.pm:
```diff
@@ -0,0 +1,310 @@
+		} elsif ($cockblock->{'TYPE'} == $COCKBLOCK_TYPE{'FINAL'} && ($prev_type == $COCKBLOCK_TYPE{'INITIAL'} || $prev_type == $COCKBLOCK_TYPE{'INTERMEDIATE'})) {
+			$condom .= $cyphallic;
+			my ($destroked, $strokecount) = $self->destroke($condom);
+			$dechoded .= $destroked;
```
```suggestion
                        $strokes = $strokecount;
			$dechoded .= $destroked;
```

fidsah commented on cpi/perl/Cocktography.pm:
```diff
@@ -0,0 +1,310 @@
+		$prev_type = $cockblock->{'TYPE'};
+	}
+	
+	return $dechoded;
```
```suggestion
	return { "message" => $dechoded, "strokes" => $count };
```

fidsah replied:
It'd be nice to return the stroke count along with the message so that the scripts can display how many strokes they were sent.  This would probably require tweaks to the unit test.

fidsah reviewed (commented):
Overall, it's  working fine in testing, but there's no way that I see to get strokes from a dechoded message.

notjeian committed:
Return stroke count from dechode_string()

Changed dechode_string() to return an array with text and stroke count. Updated unit test accordingly.
